src/View/AbstractDirectoryView.vala

```diff
@@ -3674,8 +3631,8 @@ namespace FM {
 
             var info = new GLib.FileInfo ();
             var dir = slot.directory;
-            string sort_col_s = get_string_from_column_id (sort_column_id);
-            string sort_order_s = (sort_order == Gtk.SortType.DESCENDING ? "true" : "false");
+            unowned string sort_col_s = ((FM.ListModel.ColumnID) sort_column_id).to_string ();
```

Comment:
This is not equivalent code.  Need to check how it affects restoring sort column from existing meta-data.

Reply:
It looks to provide really similar results though: https://github.com/elementary/files/blob/master/libcore/fm-list-model.c#L1334

Reply:
Ah, OK - I thought it was going to change the sort-column-id metadata to "0", "1", "2" but now I see that because of the cast it will not change the metadata.